src/util.ts

```diff
@@ -0,0 +1,33 @@
+export const fetchVstorageKeys = (
+  rpcAddr: string,
+  path: string,
+  height?: number,
+) => {
+  const options = {
+    method: 'POST',
+    body: JSON.stringify({
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'abci_query',
+      params: {
+        path: '/custom/vstorage/children/' + path,
+        height: height && height.toString(),
+      }, // height must be a string (bigint)
+    }),
+  };
+
+  return fetch(rpcAddr, options)
+    .then(res => res.json())
+    .then(
+      d => d.result.response.value && JSON.parse(atob(d.result.response.value)),
+    );
+};
+
+export const fetchRPCAddr = async (netconfigURL: string) => {
+  const response = await fetch(netconfigURL, {
+    headers: { accept: 'application/json' },
+  });
+  const networkConfig = await response.json();
+
+  return networkConfig.rpcAddrs[0];
```

Comment:
why just the first? best to round-robin, no?

should casting abstract this too?

Reply:
Just changed it to use a random selection. We could get fancier, but at least across many clients this will even the load. I think all the netconfigs only use a single rpc anyway though.

Casting does handle this, so if we could query the keys with casting we could remove this too.